src/mito2/src/read.rs

```diff
@@ -647,6 +647,17 @@ impl Source {
             Source::Iter(iter) => iter.next().transpose(),
         }
     }
+
+    /// Returns next non-empty [Batch].
+    pub(crate) async fn next_non_empty_batch(&mut self) -> Result<Option<Batch>> {
```

Comment:
`next_batch` should be enough. The return type is `Option` which includes empty and non-empty cases

Reply:
In most cases, it should be enough, but the `Batch` itself may be `empty` (no rows in the batch).

Reply:
For simplicity, I remove this method.